### DISCUSSION TITLE
how to get the curve data with pyaedt ?

### DISCUSSION THREAD

**Discussion by iaskgithub** [ansys/pyaedt#4314] [General]
![image](https://github.com/ansys/pyaedt/assets/52647687/053b388c-f024-4569-8ec7-5f1cfd87fda5)
the curves is created  in the maxwell , I want read them via the pyaedt

datax=proj.post.get_solution_data_per_variation(sweeps={"TSC_Ipeak": ["All"]}, expression="Moving1.Torque") datar=datax.data_real()
datax.plot()

I just get one curve of them ,how to get  all of them? 

 which function should i use **get_solution_data_per_variation** or **get_solution_data**?

**Comment by Samuelopez-ansys** [comment]
@iaskgithub 

Get_solution_data_per_variation, only obtains one variation per call, in your case you should use get_solution_data, if you debug, you will find how get_solution_data uses get_solution_data_per_variation at the end.

When you use get_solution_data, pyaedt returns an object, you can change the active trace inside this object, like in this example:

https://aedt.docs.pyansys.com/version/stable/examples/03-Maxwell/Maxwell3D_Team3_bath_plate.html#get-simulation-results-from-a-solved-setup

My suggestion is checking the available methods and properties doing: dir(datax)

**Reply by iaskgithub** [reply]
ok ,it's a nice example.thank you very much**@Samuelopez-ansys**
